l1-contracts/test/staking_asset_handler/setMinMintInterval.t.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.27;
+
+import {StakingAssetHandlerBase} from "./base.t.sol";
+import {StakingAssetHandler, IStakingAssetHandler} from "@aztec/mock/StakingAssetHandler.sol";
+import {Ownable} from "@oz/access/Ownable.sol";
+
+// solhint-disable comprehensive-interface
+// solhint-disable func-name-mixedcase
+// solhint-disable ordering
+
+contract SetMinMintIntervalTest is StakingAssetHandlerBase {
+  uint256 internal constant MINIMUM_MINT_INTERVAL = 1;
+  uint256 internal constant MAX_DEPOSITS_PER_MINT = 1;
+
+  function setUp() public override {
+    super.setUp();
+    stakingAssetHandler = new StakingAssetHandler(
+      address(this),
+      address(stakingAsset),
+      address(staking),
+      WITHDRAWER,
+      MINIMUM_STAKE,
+      MINIMUM_MINT_INTERVAL,
+      MAX_DEPOSITS_PER_MINT,
+      new address[](0)
+    );
+    stakingAsset.addMinter(address(stakingAssetHandler));
+  }
+
+  function test_WhenCallerOfSetMinMintIntervalIsNotOwner(address _caller) external {
+    vm.assume(_caller != address(this));
+    // it reverts
+    vm.expectRevert(abi.encodeWithSelector(Ownable.OwnableUnauthorizedAccount.selector, _caller));
+    vm.prank(_caller);
+    stakingAssetHandler.setMinMintInterval(MINIMUM_MINT_INTERVAL);
+  }
+
+  function test_WhenCallerOfSetMinMintIntervalIsOwner(uint256 _newMinMintInterval) external {
+    // it sets the min mint interval
+    // it emits a {MinMintIntervalSet} event
+    vm.expectEmit(true, true, true, true, address(stakingAssetHandler));
+    emit IStakingAssetHandler.MinMintIntervalSet(_newMinMintInterval);
+    stakingAssetHandler.setMinMintInterval(_newMinMintInterval);
+    assertEq(stakingAssetHandler.minMintInterval(), _newMinMintInterval);
+  }
+
+  function test_WhenOwnerTriesToMintBeforeTheNewIntervalHasPassed(
+    uint256 _newMinMintInterval,
+    uint256 _jump
+  ) external {
+    vm.assume(_newMinMintInterval > 0);
+    vm.assume(_jump < _newMinMintInterval);
+    vm.assume(_jump > 0);
+    stakingAssetHandler.setMinMintInterval(_newMinMintInterval);
+    // the "last mint timestamp" is 0 before the first mint
+    vm.warp(_newMinMintInterval - _jump);
+
+    // it reverts
+    vm.expectRevert(
+      abi.encodeWithSelector(
+        IStakingAssetHandler.NotEnoughTimeSinceLastMint.selector, 0, _newMinMintInterval
+      )
+    );
+    stakingAssetHandler.addValidator(address(1), address(1));
+  }
+
+  function test_WhenOwnerTriesToMintAfterTheNewIntervalHasPassed(uint256 _newMinMintInterval)
+    external
+  {
+    vm.assume(_newMinMintInterval > 0);
+    stakingAssetHandler.setMinMintInterval(_newMinMintInterval);
+    vm.warp(_newMinMintInterval);
```

Comment:
Could jump into the past 🤷 Seems like a bound and including the `block.timestamp` might be useful here.